data/pigui/game.lua

```diff
@@ -0,0 +1,539 @@
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_SPEED_OF_APPROACH_TO_TARGET, lui.HUD_SPEED_OF_APPROACH_TO_TARGET })
+
+		-- current brake distance
+		local brake_distance = player:GetDistanceToZeroV(velocity:magnitude(),"forward")
+		local brake_distance_retro = player:GetDistanceToZeroV(velocity:magnitude(),"reverse")
+		local altitude = player:GetAltitudeRelTo(target)
+		local ratio = brake_distance / altitude
+		local ratio_retro = brake_distance_retro / altitude
+		local speed, speed_unit = ui.Format.Speed(velocity:magnitude())
+
+		-- speed
+		uiPos = ui.pointOnClock(center, radius, 3.5)
+		local distance,unit = ui.Format.Distance(brake_distance)
+		ui.addFancyText(uiPos,
+										{ "~" .. distance, unit },
+										{ colorDark, colorDark },
+										{ ui.fonts.pionillium.medium, ui.fonts.pionillium.small },
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_BRAKE_DISTANCE_MAIN_THRUSTERS, lui.HUD_BRAKE_DISTANCE_MAIN_THRUSTERS })
+		-- current altitude
+		uiPos = ui.pointOnClock(center, radius, 3)
+		local altitude, altitude_unit = ui.Format.Distance(altitude)
+		ui.addFancyText(uiPos,
+										{ altitude, altitude_unit, " " .. speed, speed_unit },
+										{ colorLight, colorDark, colorLight, colorDark },
+										{ ui.fonts.pionillium.medium, ui.fonts.pionillium.small, ui.fonts.pionillium.medium, ui.fonts.pionillium.small },
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_DISTANCE_TO_SURFACE_OF_TARGET, lui.HUD_DISTANCE_TO_SURFACE_OF_TARGET, lui.HUD_SPEED_RELATIVE_TO_TARGET, lui.HUD_SPEED_RELATIVE_TO_TARGET })
+		-- current speed of approach
+		if approach_speed < 0 then
+			displayReticuleBrakeGauge(center, ratio, ratio_retro)
+		end
+	end
+	-- frame / target switch buttons
+	local uiPos = ui.pointOnClock(center, radius, 4.2)
+	local mouse_position = ui.getMousePos()
+	local size = 24
+	if combatTarget or navTarget then
+		ui.addIcon(uiPos, ui.theme.icons.display_frame, frameColor, size, ui.anchor.left, ui.anchor.bottom, lui.HUD_SHOW_FRAME)
+		if ui.isMouseClicked(0) and (mouse_position - (uiPos + Vector(size/2, -size/2))):magnitude() < size/2 then
+			reticuleTarget = "frame"
+		end
+		uiPos = uiPos + Vector(size,0)
+	end
+	if navTarget then
+		ui.addIcon(uiPos, ui.theme.icons.display_navtarget, navTargetColor, size, ui.anchor.left, ui.anchor.bottom, lui.HUD_SHOW_NAV_TARGET)
+		if ui.isMouseClicked(0) and (mouse_position - (uiPos + Vector(size/2, -size/2))):magnitude() < size/2 then
+			reticuleTarget = "navTarget"
+		end
+		uiPos = uiPos + Vector(size,0)
+	end
+	if combatTarget then
+		ui.addIcon(uiPos, ui.theme.icons.display_combattarget, combatTargetColor, size, ui.anchor.left, ui.anchor.bottom, lui.HUD_SHOW_COMBAT_TARGET)
+		if ui.isMouseClicked(0) and (mouse_position - (uiPos + Vector(size/2, -size/2))):magnitude() < size/2 then
+			reticuleTarget = "combatTarget"
+		end
+	end
+end
+
+local function displayCombatTargetIndicator(center, combatTarget)
```

Comment:
Since @fluffyfreak seems a little bit concerned about performance, how about saving a function call and only call this method with a valid combat target?